src/sentry/middleware/integrations/parsers/msteams.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import logging
+
+from django.http import HttpResponse
+from django.urls import resolve
+
+from sentry.middleware.integrations.parsers.base import BaseRequestParser
+from sentry.models.integrations.integration import Integration
+from sentry.models.outbox import WebhookProviderIdentifier
+from sentry.services.hybrid_cloud.util import control_silo_function
+from sentry.types.integrations import EXTERNAL_PROVIDERS, ExternalProviders
+
+logger = logging.getLogger(__name__)
+
+
+class MsTeamsRequestParser(BaseRequestParser):
+    provider = EXTERNAL_PROVIDERS[ExternalProviders.MSTEAMS]
+    webhook_identifier = WebhookProviderIdentifier.MSTEAMS
+
+    @control_silo_function
+    def get_integration_from_request(self) -> Integration | None:
```

Comment:
Could contain those methods in some way that tells readers why we're searching the payloads like this, that'd be helpful. E.g.

```python
integration = None
# For issue state change webhooks....
try:
    payload = data["value"]["payload"]
    integration_id = payload["integrationId"]
    ...
```
or
```python
integration = get_integration_from_payload()
if not integration:
    integration = get_integration_from_channel_data()
return integration:
```

Reply:
Actually, even better yet, if you could share this 'parsing for integration' logic with the actual webhook class, it'd make it so that if someone needs to update the integration code, this will also change to accomodate.